pymc3/step_methods/smc.py

```diff
@@ -212,15 +210,20 @@ def astep(self, q0):
             q_new = q0
 
         else:
-            if not self.stage_sample:
-                self.proposal_samples_array = self.proposal_dist(self.n_steps)
+            if not self.steps_until_tune and self.tune_interval:
 
-            if not self.steps_until_tune and self.tune:
                 # Tune scaling parameter
-                self.scaling = tune(self.accepted / float(self.tune_interval))
+                acc_rate = self.accepted / float(self.tune_interval)
+                self.scaling = tune(acc_rate)
                 # Reset counter
                 self.steps_until_tune = self.tune_interval
                 self.accepted = 0
+                self.stage_sample = 0
+
+                self.n_steps = int(max(1, np.log(0.01) / np.log(1 - acc_rate)))
```

Comment:
here will be divided by zero if the acceptance rate is 0! I think it is safer to right away put it to 1 if acceptance is 0???

Reply:
You are right but NumPy handles this and returns  `-inf` instead of an error, thus `n_steps` will be set to 1. I guess one change I should do is to use  `np.log1p(- acc_rate)` instead of np.log(1 - acc_rate)...

Reply:
I changed this a little bit to ensure that the lower the `acc_rate` the higher the number of `n_steps`, Now  `self.accepted == 0` is equivalent to `self.accepted == 1`, this is only true for the computation of `self.n_steps`, the computation of `self.scaling` is not affected.